cpp/include/copc-lib/hierarchy/key.hpp

```diff
@@ -24,8 +25,13 @@ class Box;
 class VoxelKey
 {
   public:
-    VoxelKey(int32_t d, int32_t x, int32_t y, int32_t z) : d(d), x(x), y(y), z(z) {}
+    VoxelKey(const int32_t &d, const int32_t &x, const int32_t &y, const int32_t &z) : d(d), x(x), y(y), z(z) {}
     VoxelKey() : VoxelKey(-1, -1, -1, -1) {}
+    VoxelKey(const std::vector<int32_t> &vec) : d(vec[0]), x(vec[1]), y(vec[2]), z(vec[3])
+    {
+        if (vec.size() != 4)
```

Comment:
This check should be done before you assign d,x,y,z. This is in case the vector has less than 4 elements, the program should throw our runtime_error not an out of index error.